discovery/wh_discovery.go

```diff
@@ -3,80 +3,148 @@ package discovery
 import (
 	"encoding/json"
 	"errors"
+	"fmt"
 	"io/ioutil"
+	"math/rand"
 	"net/http"
 	"net/url"
 	"time"
 
+	"github.com/ethereum/go-ethereum/common"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/livepeer/go-livepeer/core"
+	lpTypes "github.com/livepeer/go-livepeer/eth/types"
 	"github.com/livepeer/go-livepeer/net"
 
 	"github.com/golang/glog"
 )
 
+var whRefreshInterval = 1 * time.Minute
+
 type webhookResponse struct {
 	Address string
 }
 
 type webhookPool struct {
-	node     *core.LivepeerNode
-	callback *url.URL
+	db                   *DBOrchestratorPoolCache
+	callback             *url.URL
+	hashPreviousResponse ethcommon.Hash
+	lastRequested        time.Time
 }
 
 func NewWebhookPool(node *core.LivepeerNode, callback *url.URL) *webhookPool {
-	return &webhookPool{node: node, callback: callback}
+	addresses, hash, err := cacheWebhookTranscoders(node, callback, ethcommon.BytesToHash([]byte{}))
+
+	if err != nil || len(addresses) == 0 {
+		glog.Error("Error getting urls from webhook: ", err)
+		return nil
+	}
+
+	return &webhookPool{
+		db:                   NewDBOrchestratorPoolCache(node),
+		callback:             callback,
+		lastRequested:        time.Now(),
+		hashPreviousResponse: hash,
+	}
+}
+
+func (w *webhookPool) GetURLs() []*url.URL {
+	return w.db.GetURLs()
+}
+
+func (w *webhookPool) Size() int {
+	return len(w.db.GetURLs())
```

Comment:
We chatted about this on our last one on one for a bit, but to summarize what I think we concluded:
- The size mismatch is okay for now
- We should remove all DB entries that do not exist in the new set of Os pulled from the Webhook. @j0sh I'm thinking we can just loop through the saved DB Os, compare with the new O list and remove those that don't exist in the new O list, then save new O's that aren't in the DB. Does that make sense as a list of steps?

As for redundant `GetOrchInfo` calls, do you mean when we first create the webhook pool in `NewWebhookPool` and when we call `n.OrchestratorPool.GetOrchestrators(count)` in `selectOrchestrator`? Do you think this should be addressed in this PR?

(though this might be a mute point since it seems we concluded that DB caching isn't necessary for now)

Reply:
> The size mismatch is okay for now

:+1:

> I'm thinking we can just loop through the saved DB Os, compare with the new O list and remove those that don't exist in the new O list, then save new O's that aren't in the DB

Or just DROP the table / DELETE all entries from the DB unconditionally, and repopulate.

> As for redundant GetOrchInfo calls,

That would happen every time we call out to `GetOrchestrators(count)` and receive a response that leads to a DB update. This is because `GetOrchestrators` and `cacheDBOrchs` both invoke `GetOrchInfo`. But yeah, not a concern if we omit the DB caching for now.

Reply:
> Or just DROP the table / DELETE all entries from the DB unconditionally, and repopulate.

Ah yes, that's easier!

> not a concern if we omit the DB caching for now.

That makes sense, great :)